doc/python/categorical-colors.md

```diff
@@ -0,0 +1,231 @@
+---
+jupyter:
+  jupytext:
+    notebook_metadata_filter: all
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: "1.2"
+      jupytext_version: 1.3.1
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+  language_info:
+    codemirror_mode:
+      name: ipython
+      version: 3
+    file_extension: .py
+    mimetype: text/x-python
+    name: python
+    nbconvert_exporter: python
+    pygments_lexer: ipython3
+    version: 3.6.8
+  plotly:
+    description:
+      How to use and configure categorical (also known as qualitative or
+      discrete) color sequences.
+    display_as: file_settings
+    has_thumbnail: true
+    ipynb: ~notebook_demo/187
+    language: python
+    layout: base
+    name: Categorical Colors
+    order: 27
+    permalink: python/categorical-color/
+    thumbnail: thumbnail/heatmap_colorscale.jpg
+    v4upgrade: true
+---
+
+### Categorical vs Continuous Color
+
+In the same way as the X or Y position of a mark in cartesian coordinates can be used to represent continuous values (i.e. amounts or moments in time) or categories (i.e. labels), color can be used to represent continuous or categorical data. This page is about using color to represent **categorical** data, but Plotly can also [represent continuous values with color](/python/colorscales/).
+
+### Categorical Color Concepts
+
+This document explains the following categorical-color-related concepts:
+
+- **color sequences** are lists of colors to be mapped onto discrete data values. No interpolation occurs when using color sequences, unlike with [continuous color scales](/python/colorscales/), and each color is used as-is. Color sequence defaults depend on the `layout.colorway` attribute of the active [template](/python/templates/), and can be explicitly specified using the `color_discrete_sequence` argument for many [Plotly Express](/python/plotly-express/) functions.
```

Comment:
`layout.colorway`: except for pie, treemap, sunburst etc. but maybe we don't want to go into such details here.

Reply:
yeah, I think that's too much detail, and also `layout.*colorway` defaults to `layout.colorway` if it's not set IIRC.